src/plugins/auto/executable_network.cpp

```diff
@@ -686,37 +687,44 @@ InferenceEngine::Parameter MultiDeviceExecutableNetwork::GetConfig(const std::st
                 auto deviceInfo =  _loadContext[ACTUALDEVICE].deviceInfo;
                 lock.unlock();
                 unsigned int optimalBatchSize = 0;
                 unsigned int requests = 0;
-                std::map<std::string, InferenceEngine::Parameter> options;
-                options["MODEL_PTR"] = std::const_pointer_cast<ngraph::Function>(_network.getFunction());
+                bool bThroughputEnabledInPlugin = false;
                 try {
-                    optimalBatchSize = _core->GetMetric(deviceInfo.deviceName,
-                                    METRIC_KEY(OPTIMAL_BATCH_SIZE), options).as<unsigned int>();
-                    LOG_DEBUG("[AUTOPLUGIN]BATCHING:%s:%ld", "optimal batch size", optimalBatchSize);
+                    // for benchmark through AUTO:CPU,GPU
+                    // SetConfig directly set to CPU/GPU in this case
+                    bThroughputEnabledInPlugin =
+                        _core->GetConfig(deviceInfo.deviceName, CONFIG_KEY(PERFORMANCE_HINT)).as<std::string>() == CONFIG_VALUE(THROUGHPUT);
                 } catch (...) {
-                    LOG_DEBUG("[AUTOPLUGIN]BATCHING:%s", "metric OPTIMAL_BATCH_SIZE not supported");
+                    LOG_DEBUG("[AUTOPLUGIN]GetMetric:%s for %s", "PERF_HINT config not supported", deviceInfo.deviceName.c_str());
                 }
-                if (optimalBatchSize > 1) {
-                    const auto& mode = deviceInfo.config.find(CONFIG_KEY(PERFORMANCE_HINT));
+                const auto& mode = deviceInfo.config.find(CONFIG_KEY(PERFORMANCE_HINT));
+                if (bThroughputEnabledInPlugin ||
+                    (mode != deviceInfo.config.end() && mode->second == CONFIG_VALUE(THROUGHPUT))) {
+                    std::map<std::string, InferenceEngine::Parameter> options;
+                    options["MODEL_PTR"] = std::const_pointer_cast<ngraph::Function>(_network.getFunction());
                     try {
-                        // for benchmark through AUTO:CPU,GPU
-                        // SetConfig directly set to CPU/GPU in this case
-                        auto bThroughputEnabledInPlugin =
-                            _core->GetConfig(deviceInfo.deviceName, CONFIG_KEY(PERFORMANCE_HINT)).as<std::string>() == CONFIG_VALUE(THROUGHPUT);
-                        if (bThroughputEnabledInPlugin ||
-                            (mode != deviceInfo.config.end() && mode->second == CONFIG_VALUE(THROUGHPUT))) {
+                        optimalBatchSize = _core->GetMetric(deviceInfo.deviceName,
+                                        METRIC_KEY(OPTIMAL_BATCH_SIZE), options).as<unsigned int>();
+                        LOG_DEBUG("[AUTOPLUGIN]BATCHING:%s:%ld", "optimal batch size", optimalBatchSize);
+                    } catch (...) {
+                        LOG_DEBUG("[AUTOPLUGIN]BATCHING:%s", "metric OPTIMAL_BATCH_SIZE not supported");
+                    }
+                    if (optimalBatchSize > 1) {
```

Comment:
@myshevts @wangleis Maxim, I intend to use the ALLOW_AUTO_BATCHING key to help decide the optimal num request in AUTO, e.g, when user explicitly disable batching through this key, we will expose the general optimal number instead of the optimal number of batching. currently this config is been erased after used(maybe due to plugin does not support this config?). Shall we enable a new key in auto for this?
https://github.com/openvinotoolkit/openvino/blob/master/src/inference/src/ie_core.cpp#L552

Reply:
@myshevts @songbell In IE Core, batch_mode config should be passed for virtual plugin like AUTO and erased for HW device plugin like CPU or GPU. Please add additional target plugin check in IE Core.

Reply:
the ALLOW_AUTO_BATCHING is the general **disabling**  mechanism.
(it is assumed to be enabled by default and is triggered for nay device supporting the OPTIMAL_BATCH_SZIE metric)

Yes, you have to check that (maybe we shall need a simple function to reuse from the ie_core, rather than duplicate the code with the logic). However this is less critical as the only side effect is reporting too many inference requests to be optimal for the AUTO.
Even if the user created them....this is not a problem per se, only a minor memory waste, in reality the user will be limiting the #requests explicitly anyway (via PERF_HINT_NUM_REQUESTS). 
(e.g. for the benchmarking, where some of the those "too many" requests will be sitting idle, if the device underneath is actually able to process much less of them simultaneously becuase the auto-batching is not used .... that would inflate the latency, but would not harm the througput)

Reply:
If AUTO cannot aware user disabling auto batching, side effects are not only OPTIMAL_NUMBER_OF_INFER_REQUESTS but also disabling functionality in AUTO. AUTO:GPU will always enable auto batching on GPU even ALLOW_AUTO_BATCHING is no.

We expect IE Core can hold flag for user disabling even ALLOW_AUTO_BATCHING is erased. This flag can disable auto batching for all kinds of situations included AUTO:GPU. AUTO only need query this flag for OPTIMAL_NUMBER_OF_INFER_REQUESTS.

If AUTO querying IE Core flag doesn't affect the IE Core API, we will fix disabling in AUTO as our expectation in this PR. 

Otherwise, we will fix disabling in AUTO by w/a first which IE Core will set flag in AUTO before erase ALLOW_AUTO_BATCHING. We can continue to discuss the final solution and create another PR to fix w/a.